pkg/storegateway/indexheader/lazy_binary_reader.go

```diff
@@ -64,6 +65,10 @@ func NewLazyBinaryReaderMetrics(reg prometheus.Registerer) *LazyBinaryReaderMetr
 			Help:    "Duration of the index-header lazy loading in seconds.",
 			Buckets: []float64{0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 15, 30, 60, 120, 300},
 		}),
+		eagerLoadCount: promauto.With(reg).NewCounter(prometheus.CounterOpts{
```

Comment:
Wouldn't the value persist across restart? I am still thinking the counter might be useful. I also use it as a proxy in the unit test 😛 . But let me again if you really think it is better to remove this.

Reply:
no, prometheus counters are only kept in memory. So a restart resets them to 0.

I looked through the tests and I think we can just use the `lazyReader.loadCount` counter to make the same assertions. 

In TestReaderPool_NewBinaryReader if we run some assertions before calling `LabelNames`, then we can assert that there are no loaded blocks. You can add another `checkMetricFn` which is ran before calling LabelNames on the reader.

In TestNewLazyBinaryReader_EagerLoadLazyLoadedIndexHeaders you can just remove the usage of `eagerLoadCount` because its value is the same as `loadCount`.

Reply:
Updated the test as suggested and also do metric check assertion directly instead of using function call.